overlord/devicestate/handlers_serial.go

```diff
@@ -513,6 +513,16 @@ func getSerial(t *state.Task, regCtx registrationContext, privKey asserts.Privat
 	}
 
 	st := t.State()
+
+	shouldRequest, err := shouldRequestSerial(st, regCtx.GadgetForSerialRequestConfig())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if !shouldRequest {
+		return nil, nil, nil
```

Comment:
this should do t.Logf though as I asked

Reply:
Ugh, sorry. Not sure what happened there.